packages/core/src/components/AutocompleteField/AutocompleteField.jsx

```diff
@@ -0,0 +1,198 @@
+import Button from '../Button/Button';
+import Downshift from 'downshift';
+import PropTypes from 'prop-types';
+import React from 'react';
+import TextField from '../TextField/TextField';
+import uniqueId from 'lodash.uniqueid';
+
+/**
+ * Determine if a React component is a TextField
+ * @param {React.Node} child - a React component
+ * @return {Boolean} Is this a TextField component?
+ */
+function isTextField(child) {
+  return child != null && child.type === TextField;
+}
+
+/**
+ * An `AutocompleteField` component renders a typeahead input field, form label, and a clear input link.
+ */
+export class AutocompleteField extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.id = uniqueId('autocomplete_');
+  }
+
+  renderChildren(getInputProps) {
+    // Extend props on the TextField, by passing them through
+    // Downshift's `getInputProps` method
+    return React.Children.map(this.props.children, child => {
+      const { isDisabled } = this.props;
+
+      if (isTextField(child)) {
+        return React.cloneElement(
+          child,
+          getInputProps({
+            className: isDisabled ? 'ds-c-autocomplete__inner--disabled' : '',
+            disabled: isDisabled,
+            fieldClassName: 'ds-c-autocomplete__input',
+            id: this.id,
+            inputId: this.id,
+            labelClassName: 'ds-u-margin-top--0 ds-c-autocomplete__label'
```

Comment:
1. Removing the label's margin is going to cause issues with spacing when this component is used in the real world. If someone needs to remove the margin, they can do that themself through the `TextField` child's props.
2. `ds-c-autocomplete__label` doesn't appear have any styles associated with it, so we can remove it unless you have an intended usage for it?